### PR TITLE
[traffic-gen-in-vm] Add trex config files

### PR DESCRIPTION
**Pull request by RamLavi**
This PR adds config files to the trex package, that will be consumed in the checkup's traffic-gen configmap.
These files are going to be consumed by the trex VMI in later PRs.

### CODE REVIEW

File: pkg/internal/checkup/trex/configfiles.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
+package trex
```
Comment (orelmisan) [inline pkg/internal/checkup/trex/configfiles.go:1]: nit: Please add the license comment.

File: pkg/internal/checkup/trex/configfiles_test.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
+package trex_test
```
Comment (orelmisan) [inline pkg/internal/checkup/trex/configfiles_test.go:1]: nit: Please add the license comment.

File: pkg/internal/checkup/trex/configfiles.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+	"github.com/kiagnose/kubevirt-dpdk-checkup/pkg/internal/config"
+)
+
+const (
+	CfgFileName                = "trex_cfg.yaml"
+	StreamPyFileName           = "testpmd.py"
+	StreamPeerParamsPyFileName = "testpmd_addr.py"
+)
+
+type Configs struct {
```
Comment (orelmisan) [inline pkg/internal/checkup/trex/configfiles.go:15]: nit: Why plural?

**Comment by RamLavi** [comment]
```
$  TEST_NAMESPACE=dpdk-checkup-ns-1 NETWORK_ATTACHMENT_DEFINITION_NAME=dpdk-sriovnetwork-ns-1 RUNTIME_CLASS_NAME=performance-profile-1 make TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest TRAFFIC_GEN_IMAGE_URL=quay.io/kiagnose/kubevirt-dpdk-checkup-traffic-gen:traffic-gen-in-vm VM_CONTAINER_DISK_IMAGE_URL=quay.io/kiagnose/kubevirt-dpdk-checkup-vm:traffic-gen-in-vm test/e2e
docker run --rm \
           --volume /home/ralavi/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:/home/ralavi/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:Z \
           --volume /home/ralavi/.kube:/root/.kube:Z,ro \
           --workdir /home/ralavi/go/src/github.com/kiagnose/kubevirt-dpdk-checkup \
           -e KUBECONFIG=/root/.kube/kubeconfig \
           -e TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest \
           -e TEST_NAMESPACE=dpdk-checkup-ns-1 \
           -e NETWORK_ATTACHMENT_DEFINITION_NAME=dpdk-sriovnetwork-ns-1 \
           -e TRAFFIC_GEN_IMAGE_URL=quay.io/kiagnose/kubevirt-dpdk-checkup-traffic-gen:traffic-gen-in-vm \
           -e VM_CONTAINER_DISK_IMAGE_URL=quay.io/kiagnose/kubevirt-dpdk-checkup-vm:traffic-gen-in-vm \
           docker.io/library/golang:1.20.6 go test ./tests/... -test.v -test.timeout=1h -ginkgo.v -ginkgo.timeout=1h
=== RUN   TestKubevirtDpdkCheckup
Running Suite: KubevirtDpdkCheckup Suite - /home/ralavi/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests
==============================================================================================================
Random Seed: 1689765054

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/home/ralavi/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/test_suite_test.go:60
[BeforeSuite] PASSED [0.002 seconds]
------------------------------
Execute the checkup Job should complete successfully
/home/ralavi/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:79
• [143.181 seconds]
------------------------------

Ran 1 of 1 Specs in 143.184 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestKubevirtDpdkCheckup (143.18s)
PASS
ok  	github.com/kiagnose/kubevirt-dpdk-checkup/tests	143.206s
```

File: pkg/internal/checkup/trex/configfiles.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+	trafficCPUs                    string
+	numOfTrafficCPUs               string
+	portBandwidthGB                string
+	trafficGeneratorEastMacAddress string
+	trafficGeneratorWestMacAddress string
+	DPDKEastMacAddress             string
+	DPDKWestMacAddress             string
+}
+
+func NewTrexConfigs(cfg config.Config) Configs {
```
Comment (orelmisan) [inline pkg/internal/checkup/trex/configfiles.go:27]: nit:

The above function is being called like this:
```go
trex.NewTrexConfigs()
```

The `Trex` in the function name is redundant IMO.

```suggestion
func NewConfigs(cfg config.Config) Configs {
```

This comment applies to the rest of the functions bellow as well.

File: pkg/internal/checkup/trex/configfiles.go [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+		numOfTrafficCPUs:               numOfTrafficCPUs,
+		portBandwidthGB:                fmt.Sprintf("%d", cfg.PortBandwidthGB),
+		trafficGeneratorEastMacAddress: cfg.TrafficGeneratorEastMacAddress.String(),
+		trafficGeneratorWestMacAddress: cfg.TrafficGeneratorWestMacAddress.String(),
+		DPDKEastMacAddress:             cfg.DPDKEastMacAddress.String(),
+		DPDKWestMacAddress:             cfg.DPDKWestMacAddress.String(),
+	}
+}
+
+func (t Configs) GetTrexCfgFile() string {
```
Comment (orelmisan) [inline pkg/internal/checkup/trex/configfiles.go:47]: nit:
Please consider other verbs like `Generate` or `Create`.

Same for the other functions.

File: pkg/internal/checkup/configmap/configmap_test.go [hunk 6]
```diff
@@ -36,8 +36,9 @@ func TestNew(t *testing.T) {
 	name := "my-cm"
 	ownerName := "my-pod"
 	ownerUID := "1234567890"
+	data := map[string]string{}
```
Comment (orelmisan) [inline pkg/internal/checkup/configmap/configmap_test.go:39]: Could you please add a couple of keys and values to the data?

**Review by orelmisan — COMMENTED** [review]
Thank you @RamLavi.

Have you tried to run TRex with the files generated by the PR?

File: pkg/internal/checkup/trex/configfiles.go (hunk 1 above)
Comment (RamLavi) [reply pkg/internal/checkup/trex/configfiles.go:1]: DONE

File: pkg/internal/checkup/trex/configfiles.go (hunk 3 above)
Comment (RamLavi) [reply pkg/internal/checkup/trex/configfiles.go:15]: no reason. fixed

File: pkg/internal/checkup/trex/configfiles.go (hunk 4 above)
Comment (RamLavi) [reply pkg/internal/checkup/trex/configfiles.go:27]: DONE

File: pkg/internal/checkup/trex/configfiles.go (hunk 5 above)
Comment (RamLavi) [reply pkg/internal/checkup/trex/configfiles.go:47]: DONE

File: pkg/internal/checkup/trex/configfiles_test.go (hunk 2 above)
Comment (RamLavi) [reply pkg/internal/checkup/trex/configfiles_test.go:1]: DONE

**Comment by RamLavi** [comment]
[Change](https://github.com/kiagnose/kubevirt-dpdk-checkup/compare/b977b2ea4d5413cccf2a92423acd5f5a3b2974d1..15390288d4bf04e4e806a9eade746bde1da7f55c): Review fixes

File: pkg/internal/checkup/configmap/configmap_test.go (hunk 6 above)
Comment (RamLavi) [reply pkg/internal/checkup/configmap/configmap_test.go]: DONE

**Review by orelmisan — APPROVED** [review]

**Pull request merged**